usr/lib/common/pin_prompt.c

```diff
@@ -49,8 +49,8 @@ void pin_free(char **buf)
 
 const char *pin_prompt(char **buf, const char *msg)
 {
-    ssize_t n;
-    size_t s;
+    ssize_t n = 0;
```

Comment:
Assigning `n` isn't required, but assigning `s` to 0 was required for a debug build, since xlc caught a use-before-initialisation for `s` - it uses special marker values for uninitialised variable memory, and that caused `getline` to fail because it tried to `malloc` a size that was too large (~4GB, if I recall correctly). `n` was initialised for consistency with the only other variable that was declared in this function.

This caused some tests to fail otherwise, I don't believe this was a problem for core application functionality.

Reply:
Ah OK, I guess we run into some not well defined behavior. The man page of `getline()` states

_If *lineptr is set to NULL before the call, then getline() will  allocate a buffer for storing the line._
...
_Alternatively, before calling getline(), *lineptr can contain a pointer to a malloc-allocated buffer *n bytes in size.  If the buffer is not large enough to hold the line, getline() resizes it with realloc, updating *lineptr and *n as necessary._

_In either case, on a successful call, *lineptr and *n will be updated to reflect the buffer address and allocated size respectively._

What it does not tell is if *lineptr is NULL on entry, whether *n also needs to have a value set (i.e. zero). 

Certainly, setting s (which is *n inside getline()) to zero is correct in any case.